src/QueryHandle.cpp

```diff
@@ -166,10 +106,22 @@ auto QueryHandle::Execute() -> QueryStatus
     }
 
     freeResult();
-    bindParameters();
 
-    const std::string& query = (m_bind_param_count == 0) ? m_original_query : m_query_buffer;
-    if(0 == mysql_real_query(&m_mysql, query.c_str(), query.length()))
+    // ask our statement to prepare the final query string
+    m_final_statement = m_statement.prepareStatement(
+        [this](const auto& str_value) {
```

Comment:
I would use const std::string& value for the parameter instead of auto&, auto& seems weird as a parameter unless it is for some reason required